connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -122,35 +160,59 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
 object UCSingleCatalog {
   val LOAD_DELTA_CATALOG = ThreadLocal.withInitial[Boolean](() => true)
   val DELTA_CATALOG_LOADED = ThreadLocal.withInitial[Boolean](() => false)
+
+  def generateCredentialProps(
+      scheme: String,
+      temporaryCredentials: TemporaryCredentials): Map[String, String] = {
+    if (scheme == "s3") {
+      val awsCredentials = temporaryCredentials.getAwsTempCredentials
+      Map(
+        // TODO: how to support s3:// properly?
+        "fs.s3a.access.key" -> awsCredentials.getAccessKeyId,
+        "fs.s3a.secret.key" -> awsCredentials.getSecretAccessKey,
+        "fs.s3a.session.token" -> awsCredentials.getSessionToken,
+        "fs.s3a.path.style.access" -> "true",
+        "fs.s3.impl.disable.cache" -> "true",
+        "fs.s3a.impl.disable.cache" -> "true"
+      )
+    } else if (scheme == "gs") {
+      val gcsCredentials = temporaryCredentials.getGcpOauthToken
+      Map(
+        GcsVendedTokenProvider.ACCESS_TOKEN_KEY -> gcsCredentials.getOauthToken,
+        GcsVendedTokenProvider.ACCESS_TOKEN_EXPIRATION_KEY -> temporaryCredentials.getExpirationTime.toString,
+        "fs.gs.create.items.conflict.check.enable" -> "false",
```

Comment:
I think you should add this as constants in the GcsVendedTokenProvider. follow the existing template in the 2 lines above?

Reply:
I'll leave it to @bijanhoule . For abfs these keys are defined the azurebfs library, not sure if gs has something similar.

Reply:
@cloud-fan I was just talking about `GcsVendedTokenProvider.ACCESS_TOKEN_EXPIRATION_KEY ` which is defined in our file - https://github.com/unitycatalog/unitycatalog/blob/ddf5496d6c0db0e2bc8cd1bb950d641a4fadde69/connectors/spark/src/main/scala/io/unitycatalog/spark/GcsVendedTokenProvider.java#L11

so following the existing template of GCS meant adding another constant in this file.